### ISSUE TITLE
[checkpoint]: error when convert sync bn is done after checkpoint

### ISSUE THREAD

**Issue by min-xu-ai** [facebookresearch/fairscale#660]
@prigoyal found a bug when checkpoint is done before sync BN conversion, it fails with:

```
torch.nn.modules.module.ModuleAttributeError: 'SyncBatchNorm' object has no attribute '_checkpoint_fwd_counter'
```

reversing the order avoids this problem. It would be nice to detect this and tell users what's going on and how to fix it.

**Comment by anj-s** [comment]
@min-xu-ai  @prigoyal  just to confirm the steps to repro this error are:
1. call checkpoint_wrapper on a BN layer
2. convert BN to sync BN using this [API](https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html#torch.nn.SyncBatchNorm.convert_sync_batchnorm)
3. Error is raised as mentioned above

If we do 2 before 1 then the error goes away. Is this correct?

**Comment by min-xu-ai** [comment]
Yes! I think it is not limited to sync bn. Any kind of wrapper that change the module instance after checkpoint_wrapper (with `maintain_fwd_counter=True`), will have the same issue since the variable set by checkpoint_wrapper is lost.
